### PR TITLE
executor: insert ignore return errAutoincReadFailed warning

### LINKED ISSUE #38483: Unexpected Error: Failed to read auto-increment value from storage engine
## Bug Report

Unexpected Error: Failed to read auto-increment value from storage engine

### 1. Minimal reproduce step (Required)

```sql
CREATE TABLE t0(c0 SMALLINT AUTO_INCREMENT PRIMARY KEY);
INSERT IGNORE  INTO t0(c0) VALUES (194626268);

INSERT IGNORE  INTO t0(c0) VALUES ('*'); -- Error: Failed to read auto-increment value from storage engine
```

### 2. What did you expect to see? (Required)
No error.

### 3. What did you see instead (Required)
Error

### 4. What is your TiDB version? (Required)

```
| Release Version: v6.4.0-alpha-72-g3ef8352a5
Edition: Community
Git Commit Hash: 3ef8352a5754606e511ca89292a50612c289a501
Git Branch: master
UTC Build Time: 2022-10-14 12:56:17
GoVersion: go1.19.1
Race Enabled: false
TiKV Min Version: 6.2.0-alpha
Check Table Before Drop: false
Store: unistore |
```

### PR DESCRIPTION
**Pull request by hawkingrei**
Signed-off-by: Weizhen Wang <wangweizhen@pingcap.com>

### What problem does this PR solve?

Issue Number: close #38483

Problem Summary:

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: executor/insert_test.go [hunk 1]
```diff
@@ -398,6 +398,14 @@ func TestInsertWrongValueForField(t *testing.T) {
 	tk.MustExec(`SET @@sql_mode='STRICT_TRANS_TABLES'`)
 	tk.MustGetErrMsg(`INSERT INTO ts (id, time1) VALUES (2, TIMESTAMP '1018-12-24 00:00:00')`, `[table:1292]Incorrect timestamp value: '1018-12-24 00:00:00' for column 'time1' at row 1`)
 	tk.MustExec(`DROP TABLE ts`)
+
+	tk.MustExec(`CREATE TABLE t0(c0 SMALLINT AUTO_INCREMENT PRIMARY KEY);`)
+	tk.MustExec(`INSERT IGNORE  INTO t0(c0) VALUES (194626268);`)
+	tk.MustExec(`INSERT IGNORE  INTO t0(c0) VALUES ('*')`)
+	tk.MustQuery(`SHOW WARNINGS`).Check(testkit.Rows(
+		`Warning 1366 Incorrect smallint value: '*' for column 'c0' at row 1`,
+		`Warning 1690 constant 32768 overflows smallint`,
```
Comment (lance6716) [inline executor/insert_test.go:407]: seems the warning message indicates the internal behaviour is not same as MySQL, MySQL has warning of

```
MySQL [test]> show warnings;
+---------+------+-------------------------------------------------------+
| Level   | Code | Message                                               |
+---------+------+-------------------------------------------------------+
| Warning | 1366 | Incorrect integer value: '*' for column 'c0' at row 1 |
| Warning | 1062 | Duplicate entry '32767' for key 'PRIMARY'             |
+---------+------+-------------------------------------------------------+
```

So we can find another case that TiDB is different with MySQL

```
MySQL [test]> alter table t0 add column c1 int default 1;
Query OK, 0 rows affected (0.081 sec)
Records: 0  Duplicates: 0  Warnings: 0

MySQL [test]> INSERT ignore INTO t0(c0) VALUES ('*') on duplicate key update c1 = 2;
Query OK, 2 rows affected, 1 warning (0.040 sec)

MySQL [test]> select * from t0;
+-------+------+
| c0    | c1   |
+-------+------+
| 32767 |    2 |
+-------+------+
1 row in set (0.001 sec)
```

Comment (hawkingrei) [reply executor/insert_test.go:407]: It is another bug. 

Sad.

Comment (lance6716) [reply executor/insert_test.go:407]: I found this behaviour in doc

> When an AUTO_INCREMENT integer column runs out of values, a subsequent INSERT operation returns a duplicate-key error. This is general MySQL behavior.

https://dev.mysql.com/doc/refman/8.0/en/innodb-auto-increment-handling.html

**Review by lance6716 — APPROVED** [review]

File: executor/insert_test.go (hunk 1 above)
Comment (hawkingrei) [reply executor/insert_test.go:407]: OK

**Review by xhebox — APPROVED** [review]

**Pull request merged**